### PR TITLE
Bug/UAR-946-duplicate-new-oe-name

### PR DESCRIPTION
**Pull request by azafarCH**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-946

### Change description
Fixed bug where existing corporateBodyName would be populated with proposedCorporateBodyName.

```JSON
[
    {
        "data": {
            "beneficialOwnerStatement": "ALL_IDENTIFIED_ALL_DETAILS",
            "presenter": {
                "fullName": "bbhb",
                "email": "demo@ch.gov.uk"
            },
            "additions": [],
            "entityNumber": "OE111129",
            "changes": [
                {
                    "change": "changeOfEntityName",
                    "proposedCorporateBodyName": "i have changed the name"
                },
                {
                    "change": "changeOfRoa",
                    "proposedRegisteredOfficeAddress": {
                        "houseNameNum": "Level 28",
                        "street": "395 Bourke Street",
                        "area": "",
                        "postTown": "Melbourne",
                        "region": "Victoria 3000",
                        "postCode": "f",
                        "country": "Australia"
                    }
                },
                {
                    "change": "changeOfCompanyIdentification",
                    "proposedRegisterLocation": "Australia",
                    "proposedPlaceRegistered": "State Of Victoria - Australia",
                    "proposedRegisterJurisdiction": "hgguyg"
                }
            ],
            "dueDiligence": {
                "dateChecked": "2023-07-03",
                "agentName": "Kylie Ayers",
                "dueDiligenceCorrespondenceAddress": {
                    "houseNameNum": "Anjolie Hurst",
                    "street": "687 East Second Boulevard",
                    "area": "Deleniti aute volupt",
                    "postTown": "Animi odio quae mol",
                    "region": "Voluptatem rerum la",
                    "postCode": "m",
                    "country": "Northern Ireland"
                },
                "agentAssuranceCode": "Aliquam aut rerum re",
                "amlRegistrationNumber": "909",
                "email": "jikugifag@mailinator.com",
                "supervisoryBody": "Cody Frost",
                "partnerName": "Benedict Phillips",
                "diligence": "agree"
            },
            "type": "OE02",
            "payment_reference": "y2XN0zgTXCbzRcJ",
            "filingForDate": {
                "year": "2023",
                "month": "06",
                "day": "03"
            },
            "anyBOsOrMOsAddedOrCeased": true,
            "cessations": [],
            "entityName": "TESTING SERVICE ADDRESS",
            "userSubmission": {
                "entity_name": {
                    "name": "i have changed the name"
                },
                "entity_number": "OE111129",
                "presenter": {
                    "full_name": "bbhb",
                    "email": "demo@ch.gov.uk"
                },
                "entity": {
                    "serviceAddressSameAsPrincipalAddress": false,
                    "onRegisterInCountryFormedIn": true,
                    "incorporation_country": "Australia",
                    "principal_address": {
                        "property_name_number": "Level 28",
                        "line_1": "395 Bourke Street",
                        "line_2": "",
                        "town": "Melbourne",
                        "county": "Victoria 3000",
                        "country": "Australia",
                        "postcode": "f"
                    },
                    "service_address": {
                        "property_name_number": "21",
                        "line_1": "My House",
                        "line_2": "Long Street",
                        "town": "Cardiff",
                        "county": "",
                        "country": "United Kingdom",
                        "postcode": "CF14 3UZ"
                    },
                    "is_service_address_same_as_principal_address": false,
                    "email": "demo@ch.gov.uk",
                    "legal_form": "Public Company Limited By Shares",
                    "law_governed": "Corporation Law Of Australia",
                    "public_register_name": "State Of Victoria - Australia",
                    "public_register_jurisdiction": "hgguyg",
                    "registration_number": "004044937",
                    "is_on_register_in_country_formed_in": true
                },
                "due_diligence": {
                    "identity_date": "2023-07-03",
                    "name": "Kylie Ayers",
                    "identity_address": {
                        "property_name_number": "Anjolie Hurst",
                        "line_1": "687 East Second Boulevard",
                        "line_2": "Deleniti aute volupt",
                        "town": "Animi odio quae mol",
                        "county": "Voluptatem rerum la",
                        "country": "Northern Ireland",
                        "postcode": "m"
                    },
                    "email": "jikugifag@mailinator.com",
                    "supervisory_name": "Cody Frost",
                    "aml_number": "909",
                    "agent_code": "Aliquam aut rerum re",
                    "partner_name": "Benedict Phillips",
                    "diligence": "agree"
                },
                "overseas_entity_due_diligence": null,
                "beneficial_owners_statement": "ALL_IDENTIFIED_ALL_DETAILS",
                "beneficial_owners_individual": [
                    {
                        "serviceAddressSameAsUsualResidentialAddress": false,
                        "onSanctionsList": false,
                        "first_name": "Zorita",
                        "last_name": "Cash",
                        "date_of_birth": "1983-08-01",
                        "nationality": "Eritrean",
                        "second_nationality": "Lebanese",
                        "usual_residential_address": {
                            "property_name_number": "Hiram Clark",
                            "line_1": "488 Nobel Extension",
                            "line_2": "Consequatur non rem ",
                            "town": "Cupidatat ut explica",
                            "county": "Dolorem aliquam in c",
                            "country": "Paraguay",
                            "postcode": "21774"
                        },
                        "service_address": {
                            "property_name_number": "Azalia Garrison",
                            "line_1": "58 Hague Road",
                            "line_2": "Totam harum sed qui ",
                            "town": "Quo elit maiores co",
                            "county": "Qui quos velit culpa",
                            "country": "Niue",
                            "postcode": "48592"
                        },
                        "is_service_address_same_as_usual_residential_address": false,
                        "start_date": "2022-04-06",
                        "beneficial_owner_nature_of_control_types": [
                            "OVER_25_PERCENT_OF_SHARES",
                            "APPOINT_OR_REMOVE_MAJORITY_BOARD_DIRECTORS"
                        ],
                        "trustees_nature_of_control_types": [],
                        "non_legal_firm_members_nature_of_control_types": [
                            "OVER_25_PERCENT_OF_SHARES",
                            "SIGNIFICANT_INFLUENCE_OR_CONTROL"
                        ],
                        "is_on_sanctions_list": false,
                        "trust_ids": null,
                        "trust_data": null,
                        "ceased_date": "2023-04-05",
                        "ch_reference": "RandomeaP1EB70SSD9SLmiK5Y"
                    }
                ],
                "beneficial_owners_corporate": [
                    {
                        "serviceAddressSameAsPrincipalAddress": false,
                        "onRegisterInCountryFormedIn": true,
                        "onSanctionsList": false,
                        "name": "Preston Wynn",
                        "principal_address": {
                            "property_name_number": "Vera Walls",
                            "line_1": "81 White Nobel Parkway",
                            "line_2": "Ut voluptate dicta c",
                            "town": "Tempor numquam inven",
                            "county": "Soluta vitae adipisi",
                            "country": "Isle of Man",
                            "postcode": "82691"
                        },
                        "service_address": {
                            "property_name_number": "Kirsten Waters",
                            "line_1": "50 Hague Road",
                            "line_2": "Aut nemo et do sit ",
                            "town": "Autem cupiditate ali",
                            "county": "Optio do fugiat vel",
                            "country": "Aruba",
                            "postcode": "41552"
                        },
                        "is_service_address_same_as_principal_address": false,
                        "legal_form": "Eaque omnis nesciunt",
                        "law_governed": "Ullamco voluptatibus",
                        "is_on_register_in_country_formed_in": true,
                        "public_register_name": "Callum Heath",
                        "registration_number": "763",
                        "start_date": "2021-04-06",
                        "beneficial_owner_nature_of_control_types": [
                            "OVER_25_PERCENT_OF_VOTING_RIGHTS",
                            "SIGNIFICANT_INFLUENCE_OR_CONTROL"
                        ],
                        "trustees_nature_of_control_types": [],
                        "non_legal_firm_members_nature_of_control_types": [
                            "OVER_25_PERCENT_OF_SHARES",
                            "SIGNIFICANT_INFLUENCE_OR_CONTROL"
                        ],
                        "is_on_sanctions_list": false,
                        "trust_ids": null,
                        "trust_data": null,
                        "ceased_date": null,
                        "ch_reference": "RandomBoOther12345"
                    },
                    {
                        "serviceAddressSameAsPrincipalAddress": true,
                        "onRegisterInCountryFormedIn": true,
                        "onSanctionsList": true,
                        "name": "Audra Kelley",
                        "principal_address": {
                            "property_name_number": "Carly Frye",
                            "line_1": "79 East Fabien Boulevard",
                            "line_2": "Eos fugiat voluptas",
                            "town": "Aute corporis assume",
                            "county": "Modi do aut eum accu",
                            "country": "Dominica",
                            "postcode": "36993"
                        },
                        "service_address": {},
                        "is_service_address_same_as_principal_address": true,
                        "legal_form": "Fugit laborum Quis",
                        "law_governed": "Iste quaerat ut offi",
                        "is_on_register_in_country_formed_in": true,
                        "public_register_name": "Dora Sanford",
                        "registration_number": "195",
                        "start_date": "2022-04-06",
                        "beneficial_owner_nature_of_control_types": [
                            "OVER_25_PERCENT_OF_VOTING_RIGHTS",
                            "SIGNIFICANT_INFLUENCE_OR_CONTROL"
                        ],
                        "trustees_nature_of_control_types": [],
                        "non_legal_firm_members_nature_of_control_types": [
                            "APPOINT_OR_REMOVE_MAJORITY_BOARD_DIRECTORS",
                            "SIGNIFICANT_INFLUENCE_OR_CONTROL"
                        ],
                        "is_on_sanctions_list": true,
                        "trust_ids": null,
                        "trust_data": null,
                        "ceased_date": "2023-03-03",
                        "ch_reference": "RandomBoOther6789"
                    }
                ],
                "beneficial_owners_government_or_public_authority": [
                    {
                        "serviceAddressSameAsPrincipalAddress": true,
                        "onSanctionsList": true,
                        "name": "Lysandra Lee",
                        "principal_address": {
                            "property_name_number": "Darryl Huffman",
                            "line_1": "86 Old Road",
                            "line_2": "Veniam officia debi",
                            "town": "Nihil esse a archite",
                            "county": "Amet temporibus dol",
                            "country": "Taiwan",
                            "postcode": "14890"
                        },
                        "service_address": {},
                        "is_service_address_same_as_principal_address": true,
                        "legal_form": "Qui corrupti adipis",
                        "law_governed": "Animi quia expedita",
                        "start_date": "2022-04-06",
                        "beneficial_owner_nature_of_control_types": [
                            "OVER_25_PERCENT_OF_SHARES",
                            "OVER_25_PERCENT_OF_VOTING_RIGHTS",
                            "SIGNIFICANT_INFLUENCE_OR_CONTROL"
                        ],
                        "non_legal_firm_members_nature_of_control_types": [
                            "OVER_25_PERCENT_OF_VOTING_RIGHTS",
                            "SIGNIFICANT_INFLUENCE_OR_CONTROL"
                        ],
                        "is_on_sanctions_list": true,
                        "ceased_date": "2023-04-02",
                        "ch_reference": "RandomBoGov6789"
                    }
                ],
                "managing_officers_individual": [
                    {
                        "serviceAddressSameAsUsualResidentialAddress": true,
                        "first_name": "Lacy",
                        "last_name": "Lancaster",
                        "has_former_names": true,
                        "former_names": "Chloe Adkins",
                        "date_of_birth": "1988-08-01",
                        "nationality": "Swazi",
                        "second_nationality": "Honduran",
                        "usual_residential_address": {
                            "property_name_number": "Caesar Michael",
                            "line_1": "44 Green Nobel Extension",
                            "line_2": "Quia qui exercitatio",
                            "town": "Dolore velit volupta",
                            "county": "Quis veniam itaque ",
                            "country": "Sweden",
                            "postcode": "69039"
                        },
                        "service_address": {},
                        "is_service_address_same_as_usual_residential_address": true,
                        "occupation": "Et ut pariatur Dolo",
                        "role_and_responsibilities": "Reprehenderit praes",
                        "start_date": null,
                        "resigned_on": "2023-04-01",
                        "ch_reference": "app4"
                    }
                ],
                "managing_officers_corporate": [
                    {
                        "serviceAddressSameAsPrincipalAddress": true,
                        "onRegisterInCountryFormedIn": true,
                        "name": "Ignacia Robbins",
                        "principal_address": {
                            "property_name_number": "Savannah Barker",
                            "line_1": "11 East New Road",
                            "line_2": "Quasi et eligendi vo",
                            "town": "Rerum commodi minim ",
                            "county": "Sit anim aut necessi",
                            "country": "Chile",
                            "postcode": "89130"
                        },
                        "service_address": {},
                        "is_service_address_same_as_principal_address": true,
                        "legal_form": "Qui dolores sed quas",
                        "law_governed": "Aut ut veniam sit r",
                        "is_on_register_in_country_formed_in": true,
                        "public_register_name": "Irish register",
                        "registration_number": "NB1234",
                        "role_and_responsibilities": "Ullam possimus volu",
                        "contact_full_name": "Maggy Henson",
                        "contact_email": "vobukybo@mailinator.com",
                        "start_date": null,
                        "resigned_on": "2023-03-01",
                        "ch_reference": "MOcorp1"
                    }
                ],
                "trusts": [],
                "update": {
                    "date_of_creation": "2021-08-31",
                    "filing_date": "2023-06-03",
                    "bo_mo_data_fetched": true,
                    "registrable_beneficial_owner": true,
                    "no_change": false,
                    "review_beneficial_owners_individual": null,
                    "review_beneficial_owners_corporate": null,
                    "review_beneficial_owners_government_or_public_authority": null,
                    "review_managing_officers_individual": null,
                    "review_managing_officers_corporate": null
                },
                "links": {
                    "self": "/transactions/191726-643116-904532/overseas-entity/64c244f9ff83a84e04f1cf65"
                }
            },
            "payment_method": "credit-card"
        },
        "description": "Overseas entity update statement made 27 July 2023",
        "description_identifier": "**Overseas Entities** submission made",
        "description_values": {},
        "kind": "overseas-entity-update",
        "cost": "120.00"
    }
]
```

### Work checklist

- [x] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java [hunk 1]
```diff
@@ -209,6 +210,11 @@ private void collectUpdateSubmissionData(UpdateSubmission updateSubmission,
       updateSubmission.getChanges()
               .addAll(overseasEntityChangeService.collateOverseasEntityChanges(publicPrivateOeData,
                       submissionDto, logMap));
+
+      var existingCompanyName = new EntityNameDto();
```
Comment (RobertBromfield) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java:214]: You might want to run a formatter here this doesn't look properly indented

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java [hunk 2]
```diff
@@ -371,6 +375,13 @@ void testFilingGenerationWhenSuccessfulWithoutTrustsAndWithIdentityChecksForUpda
         assertEquals("OE111229", filing.getData().get("entityNumber"));
         assertEquals("OE02", filing.getData().get("type"));
 
+        final OverseasEntitySubmissionDto submissionDto = (OverseasEntitySubmissionDto)filing.getData().get("userSubmission");
```
Comment (RobertBromfield) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java:378]: I know this is following the pattern already established in the file, but we shouldn't be doing an unchecked cast.

Also same with others below the variable type isn't necessary here and it's usually preferred to use the Local Variable Type Inference (`var`) where possible.

**Review by RobertBromfield — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java [hunk 3]
```diff
@@ -209,6 +210,11 @@ private void collectUpdateSubmissionData(UpdateSubmission updateSubmission,
...
+        existingCompanyName.setName(publicPrivateOeData.getLeft().getCompanyName());
+        updateSubmission.getUserSubmission().setEntityName(existingCompanyName);
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java:216]: Does this not overwrite the entity name in the user submission? What if the user is requesting a name change? This  will reset the changed name to the old name from the API.
Suggest a new filed in the updateSubmission (called entityName) which we set to be publicPrivateOeData.getLeft().getCompanyName().
This new field will have to be picked up on the XML CHIPS consumer side.

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java (hunk 2 above)
Comment (RobertBromfield) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java:378]:
```suggestion
        assertInstanceOf(OverseasEntitySubmissionDto.class, filing.getData().get("userSubmission"));
        final var submissionDto = (OverseasEntitySubmissionDto)filing.getData().get("userSubmission");
```

**Review by RobertBromfield — DISMISSED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java (hunk 3 above)
Comment (bantao-ch) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java:216]: I think we are doing this after the name change calculation -- would that be ok ?
@azafarCH -- check if we are doing this after ....

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java:216]: Would still not be correct to change the user submission which we may pass in XML or log. Suggested adding a new (JSON) field to the UpdateSubmssion Dto and just setting that to the name from public api. New field needs to be picked up by CHIPS filing consumer and put in XML.

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java [hunk 4]
```diff
@@ -371,6 +375,13 @@ void testFilingGenerationWhenSuccessfulWithoutTrustsAndWithIdentityChecksForUpda
...
+        assertEquals("Test Public Company", submissionDto.getEntityName().getName());
+
+        final List<Change> changesList = (List<Change>)filing.getData().get("changes");
+        final EntityNameChange entityNameChange = (EntityNameChange)changesList.get(0);
```
[the remark this answers is not in the document]
Comment (azafarCH) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java:379]: can't seem to check the cast for the list

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java (hunk 3 above)
Comment (azafarCH) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java]: We've made changes, added the new field and mapped @AndreKramer7654

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java [hunk 5]
```diff
@@ -92,6 +96,14 @@ public void setEntityNumber(String entityNumber) {
         this.entityNumber = entityNumber;
     }
 
+    public String getExistingEntityName() {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:99]: This needs a @JsonProperty(ENTITY_NAME)

Can we name this field just entityName (drop existing). 
So public static final String ENTITY_NAME = "entityName";

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:99]: By "needs" - to be consistent with other decorated fields.

Comment (azafarCH) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:99]: ah okay, did not want to get it mixed up with the other entity_name field in userSubmission but if that's preferred i'll make the change

Comment (azafarCH) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:99]: will make the change on xml side too

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndreKramer7654) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java:379]: could use List<?> but it's fine I think.

**Review by AndreKramer7654 — APPROVED** [review]

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (RobertBromfield) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java:379]: It does seem to be quite difficult to do it with a List however you can still use the Local Variable Type Inference with a check on line 382 

As for line 381 I'd suggest at minimum checking it's a `List` then using `@SuppressWarnings("unchecked")` to get rid of the warning

**Review by bantao-ch — APPROVED** [review]

**Review by bbell-companieshouse — APPROVED** [review]

**Pull request merged**